### DISCUSSION TITLE
[A-L] (2023/24) Foglio 1 - Esercizio 1

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/algebra#156] [Esercizi - Viaggi/Piazza]
![es 1](https://github.com/Elia-Belli/algebra2324/assets/107944476/76af3c9b-4119-4e3a-ab90-94ee7a1ccd21)

**Comment by FeddyLix17** [comment]
ringrazio ancora @ph04 per la spiegazione e @CuriousCI per aver chiesto conferma al professore di essa.

*premessa*: in questa tipologia specifica di caso (ovvero quando entrambe le relazioni sono definite rispettivamente su due singoli insiemi diversi), definendo oltre ad $\Large A$ e $\Large B$ un insieme $\Large C = \lbrace A \cap B \rbrace =  \lbrace b, c  \rbrace $, la relazione $\Large R$ sarà definita su $\Large A \times C$ (invece che su $\Large A \times A$), mentre $\Large S$ su  $\Large C \times B$ (invece che su $\Large B \times B$).

La situazione attuale è la seguente

$$
    \Large \text{dom(R)} = A
$$

$$
     \Large \text{codom(R)} = \text{dom(S)} = C
$$

$$
     \Large \text{codom(S)} = B
$$

In conclusione

$$
     \Large R \circ S = \lbrace (x,z), \ x \in A\ \wedge\ z \in B \ |\ \exists y \in C\ :\ xSy\ \wedge\ yRz \rbrace 
$$

$$
     \Large R \circ S = \lbrace (c,c) \rbrace 
$$

---

$$
     \Large S \circ R = \lbrace (x,z),\ x \in A\ \wedge\ z \in B \ |\ \exists y \in C\ :\ xRy\ \wedge\ ySz \rbrace 
$$

$$
     \Large S \circ R = \lbrace (a,c), (a,d), (a,b), (b,b), (b,c) \rbrace 
$$

**Comment by Frisbon** [comment]
![image](https://github.com/sapienzastudentsnetwork/algebra2324/assets/84731341/61a37390-eb65-414a-be48-c70f9e74e5e1)

Soluzione coincide con @FeddyLix17 (anche se i procedimenti sono molto piú "generici")
